diagnostic_rt_qpcr/operation_types/aliquot_positive_template/protocol.rb

```diff
@@ -1,170 +1,162 @@
+  VOL_SUSPENSION = { qty: 30, units: MICROLITERS }.freeze
+  OUTPUT_ITEMS_NUM = { qty: 33, units: TUBE_MICROFUGE }.freeze
 
   def main
     # 1. Get 33 1.5 mL tube for each operation
-    get_tubes(operations.length)
+    get_tubes(count: operations.length)
     operations.retrieve
 
     # 2. For each Lyophilized Postive Control, resuspend in 1 mL of
     # nuclease-free waterand, aliquot 33 use aliquots (approximately
-    # 30 L) and store at less than and equal to -70C.
+    # 30 µL) and store at less than and equal to -70C.
+    save_output(ops: operations)
+
+    suspend_lyophilized_RNA
+    keep_tubes = [] # Empty array for storing single use aliquots
 
     # Group the operations by the input reagent
-    ops_by_input = operations.group_by {|op| op.input("Template").item}
-    ops_by_input.each do |lyophilized_RNA, ops|
-      label_and_locate_output(ops)
+    ops_by_input = operations.group_by { |op| op.input('Template').item }
+    ops_by_input.each do |lyophilized_rna, ops|
+      keep_tubes.push(make_aliquots(ops: ops, lyophilized_rna: lyophilized_rna))
     end
 
-    suspend_lyophilized_RNA(operations)
-    keep_tubes = [] #Empty array for storing single use aliquots
-    
-    ops_by_input.each do |lyophilized_RNA, ops|
-      keep_tubes.push(make_aliquots(ops))
-    end
-    
-    prepare_plating(keep_tubes)
-    
+    prepare_plating(keep_tubes: keep_tubes)
+
     # 3. Thaw a single aliquot of diluted positive control for each
     # experiment and hold on ice until adding to plate.
     # Discard any unused portion of the aliquot.
-    operations.store(interactive: true, io: "output", method: "boxes")
+    operations.store(interactive: true, io: 'output', method: 'boxes')
   end
 
   # Get 33 1.5 mL tubes per dried positive control
   #
   # @param count [Integer] the number of operations
-  def get_tubes(count)
+  def get_tubes(count:)
     show do
-      title "Get new 1.5 mL tubes"
-      check "Please get #{count*33} 1.5 mL tubes"
+      title "Get new #{TUBE_MICROFUGE}"
+      check "Please get #{count * OUTPUT_ITEMS_NUM[:qty]} #{TUBE_MICROFUGE}"
     end
   end
 
   # Label the tubes so that the same reagents have consecutive IDs
   # And move the output tubes to the right storage locations
-  #
   # @param operations [OperationList] The list of operations
-  def label_and_locate_output(operations)
-    operations.make
-
-    #Declare references to output objects
-    operations.each do |op|
-      op.output("Template").item.associate :volume, 30
-      op.output("Template").item.save
-      
-      output_RNA = op.output("Template").sample
-      for i in 0..31
-        new_aliquot = output_RNA.make_item("Purified RNA in 1.5 mL tube")
-        new_aliquot.associate :volume, 30
-        new_aliquot.save
-        # Associated output RNA with a new location wizard for storage of
-        # plasmids and fragments, M80P (for DAMP)
-        link_output_item(op, output_RNA, new_aliquot)
+  def save_output(ops:)
+    ops.make
```

Comment:
The way this is written it is probably better to call make on each op individually, otherwise the numbering of the items will be a little weird.

Reply:
But you may not want to do make at all, and just initialize all 33 items together. We can get rid of the output for this operation.